### ISSUE TITLE
CPUThrottlingHigh alert for metrics-server-nanny (addon-resizer:1.8.11-gke.0)

### ISSUE THREAD

**Issue by pgier** [kubernetes/autoscaler#4141]
**Which component are you using?**: addon-resizer and metrics-server

**What version of the component are you using?**: 1.8.11-gke.0

Component version: 1.8.11-gke.0

**What k8s version are you using (`kubectl version`)?**:

<details><summary><code>kubectl version</code> Output</summary><br><pre>
$ kubectl version
Client Version: version.Info{Major:"1", Minor:"21", GitVersion:"v1.21.1", GitCommit:"5e58841cce77d4bc13713ad2b91fa0d961e69192", GitTreeState:"clean", BuildDate:"2021-05-12T14:18:45Z", GoVersion:"go1.16.4", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"19+", GitVersion:"v1.19.9-gke.1900", GitCommit:"008fd38bf3dc201bebdd4fe26edf9bf87478309a", GitTreeState:"clean", BuildDate:"2021-04-14T09:22:08Z", GoVersion:"go1.15.8b5", Compiler:"gc", Platform:"linux/amd64"}
WARNING: version difference between client (1.21) and server (1.19) exceeds the supported minor version skew of +/-1

</pre></details>

**What environment is this in?**:

GKE version  1.19.9-gke.1900 

**What did you expect to happen?**:

Expected no Alert from this component.

**What happened instead?**:

Received the Alert, and the metrics seem to indicate that throttling is regularly over 90%.

**How to reproduce it (as minimally and precisely as possible)**:

Create basic GKE cluster and install the kube-prometheus monitoring stack.

**Anything else we need to know?**:

Could be related to issue #3833

The metrics-server-nanny logs look fine, however the metrics-server in the same pod has a lot of errors in the logs saying that metrics cannot be collected.

```
I0614 02:22:57.188713       1 log.go:172] http: TLS handshake error from 10.128.0.15:33630: EOF
I0614 02:22:57.192135       1 log.go:172] http: TLS handshake error from 10.128.0.14:36984: EOF
I0614 02:22:57.389307       1 log.go:172] http: TLS handshake error from 10.128.0.14:36960: EOF
I0614 02:22:57.988045       1 log.go:172] http: TLS handshake error from 10.68.3.1:52464: EOF
I0614 02:22:57.988154       1 log.go:172] http: TLS handshake error from 10.68.3.1:52474: EOF
E0614 02:22:58.288218       1 reststorage.go:160] unable to fetch pod metrics for pod kube-system/event-exporter-gke-67986489c8-vmg4j: no metrics known for pod
E0614 02:22:58.288278       1 reststorage.go:160] unable to fetch pod metrics for pod kube-system/kube-dns-6c7b8dc9f9-sn7p5: no metrics known for pod
E0614 02:22:58.288287       1 reststorage.go:160] unable to fetch pod metrics for pod kube-system/fluentbit-gke-wxqcn: no metrics known for pod
E0614 02:22:58.288293       1 reststorage.go:160] unable to fetch pod metrics for pod kube-system/kube-proxy-gke-paul-test-1-default-pool-1166a98e-mx9n: no metrics known for pod
E0614 02:22:58.288299       1 reststorage.go:160] unable to fetch pod metrics for pod kube-system/stackdriver-metadata-agent-cluster-level-68ffcbb78c-sqwtn: no metrics known for pod
E0614 02:22:58.288305       1 reststorage.go:160] unable to fetch pod metrics for pod kube-system/pdcsi-node-kfnfj: no metrics known for pod
E0614 02:22:58.288312       1 reststorage.go:160] unable to fetch pod metrics for pod kube-system/kube-proxy-gke-paul-test-1-default-pool-1166a98e-3zkp: no metrics known for pod
```

I created a separate issue for the metrics-server: https://github.com/kubernetes-sigs/metrics-server/issues/783

**Comment by LucasRouckhout** [comment]
We are seeing the same issue on GKE. > 90% CPU throttling on the metric server pod and same logs as you have posted. Will let you know if I discover anything. This sounds familiar to a known issue surrounding CFS quotas.

https://github.com/kubernetes/kubernetes/issues/67577
https://bugzilla.kernel.org/show_bug.cgi?id=198197

**Comment by gboor** [comment]
I just ran into this as well. For now I've disabled the alert, but it would be nice to get some sort of fix. Strangely this suddenly started happening on a cluster that is over a year old. 8 hours ago the metrics-server restarted (same version) and these alerts started popping up... no idea why it didn't happen before.

**Comment by obataku** [comment]
I came across this a couple months ago and I believe the problem should be resolved by https://github.com/kubernetes/autoscaler/pull/4112 (which hasn't landed in a versioned release yet); if you manage your own `metrics-server` deployment then you can opt to downgrade and/or use a prerelease build of addon-resizer for `metrics-server-nanny` but those using GKE will need to wait

**Comment by bskiba** [comment]
Release 1.8.14 has just come out that contains the fix for both  #3833 and #4112: https://github.com/kubernetes/autoscaler/releases/tag/addon-resizer-1.8.14

Could you verify if this fixes the issue with CPU?

**Comment by tanvp112** [comment]
@bskiba, I tried out 1.18.4 and it doesn't seem to work #4307. Can you help to take a look?
